### PR TITLE
use Safearea layout guide for ios 11 bottom constraints

### PR DESCRIPTION
**Pull request by pagrit**

### CODE REVIEW

File: BPForms/BPFormViewController.m [hunk 1]
```diff
@@ -182,13 +182,23 @@ - (void)setupTableView {
...
-        make.height.equalTo(self.view.mas_height);
-        make.top.equalTo(self.view.mas_top);
-        make.left.equalTo(self.view.mas_left);
-    }];
+
+    if (@available(iOS 11, *)) {
+        self.tableView.translatesAutoresizingMaskIntoConstraints = NO;
+        [NSLayoutConstraint activateConstraints:@[
+             [self.tableView.widthAnchor constraintEqualToAnchor:self.view.widthAnchor],
+             [self.tableView.topAnchor constraintEqualToAnchor:self.view.topAnchor],
```
Comment (soberman) [inline BPForms/BPFormViewController.m]: Why not to use `safeAreaLayoutGuide.topAnchor` as well?

File: BPForms/BPFormViewController.m [hunk 2]
```diff
@@ -182,13 +182,23 @@ - (void)setupTableView {
...
+             [self.tableView.leftAnchor constraintEqualToAnchor:self.view.leftAnchor],
+             [self.tableView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor],
+        ]];
+    }else{
```
Comment (mathaeus) [inline BPForms/BPFormViewController.m]: Spacing.

File: BPForms/BPFormViewController.m [hunk 3]
```diff
@@ -182,13 +182,23 @@ - (void)setupTableView {
...
-    
-    [self.tableView mas_makeConstraints:^(MASConstraintMaker *make) {
-        make.width.equalTo(self.view.mas_width);
-        make.height.equalTo(self.view.mas_height);
-        make.top.equalTo(self.view.mas_top);
-        make.left.equalTo(self.view.mas_left);
-    }];
+
+    if (@available(iOS 11, *)) {
+        self.tableView.translatesAutoresizingMaskIntoConstraints = NO;
```
Comment (mathaeus) [inline BPForms/BPFormViewController.m]: Please check if it would be possible to use Masonry here.

**Review by mathaeus — APPROVED** [review]

**Comment by soberman** [comment]
I don't know why my comment weren't posted... Anyways, why didn't you use top constraint with `safeAreaLayoutGuide` as well?

File: BPForms/BPFormViewController.m (hunk 1 above)
Comment (pagrit) [reply BPForms/BPFormViewController.m]: I didn't think about that. I hadn't put them anywhere.

**Comment by fabb** [comment]
+ left and right safearea constraints

**Comment by soberman** [comment]
And ^ as well, thanks @fabb

**Comment by mathaeus** [comment]
_deleted comment_

**Pull request merged**